src/ucp/core/ucp_worker.c

```diff
@@ -610,6 +655,7 @@ void ucp_worker_destroy(ucp_worker_h worker)
     ucs_trace_func("worker=%p", worker);
     ucp_worker_remove_am_handlers(worker);
     ucp_worker_destroy_eps(worker);
+    ucs_mpool_cleanup(&worker->am_mp, 1);
```

Comment:
looks like memory pool is not always created, so how come it's always destroyed?